js/src/ui/Form/TypedInput/typedInput.js

```diff
@@ -61,8 +67,11 @@ export default class TypedInput extends Component {
   };
 
   componentWillMount () {
-    if (this.props.isEth && this.props.value) {
-      this.setState({ isEth: true, ethValue: fromWei(this.props.value) });
+    const { isEth, value } = this.props;
+
+    if (typeof isEth === 'boolean' && value) {
```

Comment:
Ok, then my only grumble is the tri-state `isEth` (`null` is the default value).